### ISSUE TITLE
crash with 16bit png as input

### ISSUE THREAD

**Issue by darix** [strukturag/libheif#179]
```
$ heif-enc -b 8 -o t.heif DSCF0990.png
```

```
Program terminated with signal SIGSEGV, Segmentation fault.
#0  std::_Rb_tree<heif_channel, std::pair<heif_channel const, heif::HeifPixelImage::ImagePlane>, std::_Select1st<std::pair<heif_channel const, heif::HeifPixelImage::ImagePlane> >, std::less<heif_channel>, std::allocator<std::pair<heif_channel const, heif::HeifPixelImage::ImagePlane> > >::find (
    __k=<synthetic pointer>: heif_channel_Y, this=0x58) at /usr/include/c++/9/bits/stl_tree.h:2567
2567	    _Rb_tree<_Key, _Val, _KeyOfValue, _Compare, _Alloc>::
(gdb) bt
#0  std::_Rb_tree<heif_channel, std::pair<heif_channel const, heif::HeifPixelImage::ImagePlane>, std::_Select1st<std::pair<heif_channel const, heif::HeifPixelImage::ImagePlane> >, std::less<heif_channel>, std::allocator<std::pair<heif_channel const, heif::HeifPixelImage::ImagePlane> > >::find (
    __k=<synthetic pointer>: heif_channel_Y, this=0x58) at /usr/include/c++/9/bits/stl_tree.h:2567
#1  std::map<heif_channel, heif::HeifPixelImage::ImagePlane, std::less<heif_channel>, std::allocator<std::pair<heif_channel const, heif::HeifPixelImage::ImagePlane> > >::find (__x=<synthetic pointer>: heif_channel_Y, this=0x58) at /usr/include/c++/9/bits/stl_map.h:1194
#2  heif::HeifPixelImage::get_width (this=0x0, channel=heif_channel_Y) at heif_image.cc:189
#3  0x00007fc17b8d4642 in heif::HeifContext::Image::encode_image_as_hevc (this=0x55b4e5ee2440, image=..., encoder=0x55b4e5ee1e80, options=0x55b4e5ee2290, 
    input_class=heif_image_input_class_normal) at heif_context.cc:1682
#4  0x00007fc17b8cd97c in heif::HeifContext::encode_image(std::shared_ptr<heif::HeifPixelImage>, heif_encoder*, heif_encoding_options const*, heif_image_input_class, std::shared_ptr<heif::HeifContext::Image>&) [clone .constprop.0] (this=0x55b4e5ee16e0, pixel_image=..., encoder=0x55b4e5ee1e80, options=0x55b4e5ee2290, 
    out_image=..., input_class=<optimized out>) at heif_context.cc:1634
#5  0x00007fc17b8db27e in heif_context_encode_image (ctx=0x55b4e5ee16b0, input_image=<optimized out>, encoder=<optimized out>, options=<optimized out>, 
    out_image_handle=0x7ffda45793d8) at /usr/include/c++/9/ext/atomicity.h:96
#6  0x000055b4e5d52781 in main (argc=<optimized out>, argv=<optimized out>) at heif_enc.cc:1073
```

If a "bt full" is wanted I can provide that too.

**Comment by fancycode** [comment]
This should be fixed with https://github.com/strukturag/libheif/commit/4e62dae2abec8e2a66484c07c8317c4d11ca9a11

Saving HDR images as 8-bit HEIF is currently not supported.

**Comment by darix** [comment]
Well I would want to save 16bit into 10 or 12 bit... but that failed similarly.

**Comment by fancycode** [comment]
Saving as 10 or 12 bit is working now that https://github.com/strukturag/libheif/issues/135 has been resolved.

**Comment by darix** [comment]
is it in a released version already or do I have to pull from git for this?

**Comment by cryptomilk** [comment]
I'm not able to save do 16bit heic using darktable with libheif 1.6.0. It fails to write the data to disk.

https://github.com/cryptomilk/darktable/tree/master-heif

**Comment by fancycode** [comment]
> is it in a released version already or do I have to pull from git for this?

It is not released yet, was just fixed earlier today.

**Comment by fancycode** [comment]
> I'm not able to save to 16bit heic using darktable with libheif 1.6.0. It fails to write the data to disk.

Does it still happen with the fix from #135? If so, please open a new issue and provide more information what exactly happens and what functions you are calling.

**Comment by darix** [comment]
we opened #183  for issues with the current master branch and 16 bit images.
